mlflow/projects/__init__.py

```diff
@@ -258,10 +257,20 @@ def _fetch_git_repo(uri, version, dst_dir, git_username, git_password):
         repo.heads.master.checkout()
 
 
-def _get_conda_env_name(conda_env_path):
-    with open(conda_env_path) as conda_env_file:
-        conda_env_hash = hashlib.sha1(conda_env_file.read().encode("utf-8")).hexdigest()
-    return "mlflow-%s" % conda_env_hash
+def _get_conda_env_path(project_dir):
+    project = _load_project(project_dir)
+    if project.conda_env:
+        return os.path.abspath(os.path.join(project_dir, project.conda_env))
+    return None
+
+
+def _get_conda_env_name(project_dir):
+    project = _load_project(project_dir)
+    conda_env_contents = ""
+    if project.conda_env:
+        with open(os.path.join(project_dir, project.conda_env)) as conda_env_file:
+            conda_env_contents = conda_env_file.read()
```

Comment:
Does this work if the project does not declare a `conda_env` but does contain a `conda.yaml`? We want that to work too. We don't want to force users to create an MLproject file.

Reply:
Ah gotcha, it doesn't work currently in the case you described - will fix it

Reply:
To clarify - we do need an MLproject file because it contains the project definition (entry points etc) right? But we'll default to looking for conda.yaml in the same directory as the MLproject file if no `conda_env` is specified

Reply:
Actually - since we allow running entry points not specified in MLproject files I guess we don't need to require the MLproject either (we currently do)